File: src/server/Adfnet.Core/ValueObjects/ChildMenu.cs
using System;
using System.Collections.Generic;

namespace Adfnet.Core.ValueObjects
{
    public class ChildMenu
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Icon { get; set; }
        public string Description { get; set; }
        public virtual ICollection<LeafMenu> LeafMenus { get; set; }
        public virtual RootMenu Parent { get; set; }
    }
}

File: src/client/web/private/src/app/models/child-menu.ts
import { RootMenu } from './root-menu';
import { LeafMenu } from './leaf-menu';

export class ChildMenu {
    id: string;
    code: string;
    name: string;
    address: string;
    icon: string;
    description: string;
    parent: RootMenu;
    leafMenus: LeafMenu[];
    constructor() {
        this.parent = new RootMenu();
    }
}
